services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx

```diff
@@ -127,26 +127,79 @@ export const calculateDistanceToUpstream = (envs: Environment[]): EnvSortOrder =
     return distanceToUpstream;
 };
 
-export const EnvironmentList: React.FC<{ envs: Environment[]; release: Release; app: string; className?: string }> = ({
-    envs,
-    release,
-    app,
-    className,
-}) => {
-    const allEnvs: Environment[] = useOverview((x) => Object.values(x.environments));
-    const envPrioMap: EnvPrioMap = calculateEnvironmentPriorities(allEnvs);
+export const AppLock: React.FC<{
+    env: Environment;
+    app: string;
+    lock: Lock;
+    className?: string;
+}> = ({ env, app, lock, className }) => {
+    const deleteAppLock = useCallback(() => {
+        addAction({
+            action: {
+                $case: 'deleteEnvironmentApplicationLock',
+                deleteEnvironmentApplicationLock: { environment: env.name, application: app, lockId: lock.lockId },
+            },
+        });
+    }, [app, env.name, lock.lockId]);
     return (
-        <ul className={classNames('release-env-list', className)}>
-            {sortEnvironmentsByUpstream(envs).map((env) => (
-                <li key={env.name} className={classNames('env-card', className)}>
-                    <div className="env-card-header">
-                        <Chip
-                            className={'release-environment'}
-                            label={env.name}
-                            key={env.name}
-                            priority={envPrioMap[env.name]}
-                        />
-                        <div className={classNames('_env-card-label', className)}>
+        <Tooltip
+            key={lock.lockId}
+            arrow
+            title={'Lock Message: "' + lock.message + '" | ID: "' + lock.lockId + '"  | Click to unlock. '}
+            onClick={deleteAppLock}>
+            <div>
+                <Button icon={<Locks className="env-card-app-lock" />} className={'button-lock'} />
+            </div>
+        </Tooltip>
+    );
+};
+
+export const EnvironmentListItem: React.FC<{
+    env: Environment;
+    envPrioMap: EnvPrioMap;
+    app: string;
+    release: Release;
+    className?: string;
+}> = ({ env, envPrioMap, app, release, className }) => {
+    const deploy = useCallback(
+        () =>
+            addAction({
+                action: {
+                    $case: 'deploy',
+                    deploy: {
+                        environment: env.name,
+                        application: app,
+                        version: release.version,
```

Comment:
Here release.version is undefined

Reply:
Seems to work for me. It's either an edge case I'm missing, or I suspect it's an issue with the `repository_checkedout` directory